plugins/copr.py

```diff
@@ -217,16 +279,25 @@ def _list_installed_repositories(self, directory, enabled_only, disabled_only):
 
         for root, dir, files in os.walk(directory):
             for file_name in files:
-                if re.match("^_copr_", file_name):
+                match_new = re.match("^copr:" + self.copr_hostname, file_name)
+                match_old = self.copr_url == self.default_url and re.match("^_copr_", file_name)
+                match_any = re.match("^copr:|^_copr_", file_name)
+
+                if self.opts.hub and (match_new or match_old) or not self.opts.hub and match_any:
                     parser.read(directory + '/' + file_name)
 
         for copr in parser.sections():
             enabled = parser.getboolean(copr, "enabled")
             if (enabled and disabled_only) or (not enabled and enabled_only):
                 continue
 
-            copr_name = copr.split('-', 1)
-            msg = copr_name[0] + '/' + copr_name[1]
+            if re.match("^copr:", copr):
+                copr_name = copr.rsplit(':', 2)
+                copr_hostname = copr_name[0].split(':', 1)[1]
+                msg = copr_hostname + '/' + copr_name[1] + '/' + copr_name[2]
+            else:
+                copr_name = copr.split('-', 1)
+                msg = self.default_hostname + '/' + copr_name[0] + '/' + copr_name[1]
```

Comment:
Here the 'self.default_hostname' might be wrong, because the `copr` can come from `copr:copr.devel.redhat.com:user:project.repo`, and still we print the default_hostname (fedoraproject.org).  That's because we are loosing the context about the original filename.

Please consider this patch on top of yours please.

```patch
--- /tmp/old.py 2018-08-06 10:27:13.125289239 +0200
+++ /tmp/new.py 2018-08-06 10:52:17.475963068 +0200
@@ -274,17 +274,20 @@
             raise dnf.exceptions.Error(
                 _('Unknown subcommand {}.').format(subcommand))
 
-    def _list_installed_repositories(self, directory, enabled_only, disabled_only):
-        parser = ConfigParser()
 
-        for root, dir, files in os.walk(directory):
-            for file_name in files:
-                match_new = re.match("^copr:" + self.copr_hostname, file_name)
-                match_old = self.copr_url == self.default_url and re.match("^_copr_", file_name)
-                match_any = re.match("^copr:|^_copr_", file_name)
+    def _list_repo_file(self, directory, file_name, enabled_only, disabled_only):
+        match_new = re.match("^copr:" + self.copr_hostname, file_name)
+        match_old = self.copr_url == self.default_url and re.match("^_copr_", file_name)
+        match_any = re.match("^copr:|^_copr_", file_name)
+
+        if self.opts.hub:
+            if not (match_new or match_old):
+                return
+        elif not match_any:
+            return
 
-                if self.opts.hub and (match_new or match_old) or not self.opts.hub and match_any:
-                    parser.read(directory + '/' + file_name)
+        parser = ConfigParser()
+        parser.read(directory + '/' + file_name)
 
         for copr in parser.sections():
             enabled = parser.getboolean(copr, "enabled")
@@ -295,6 +298,10 @@
                 copr_name = copr.rsplit(':', 2)
                 copr_hostname = copr_name[0].split(':', 1)[1]
                 msg = copr_hostname + '/' + copr_name[1] + '/' + copr_name[2]
+            elif re.match("^copr:", file_name):
+                copr_name = copr.split('-', 1)
+                hostname = file_name.split(':')[1]
+                msg = hostname + '/' + copr_name[0] + '/' + copr_name[1]
             else:
                 copr_name = copr.split('-', 1)
                 msg = self.default_hostname + '/' + copr_name[0] + '/' + copr_name[1]
@@ -302,6 +309,11 @@
                 msg += " (disabled)"
             print(msg)
 
+    def _list_installed_repositories(self, directory, enabled_only, disabled_only):
+        for root, dir, files in os.walk(directory):
+            for file_name in files:
+                self._list_repo_file(directory, file_name, enabled_only, disabled_only)
+
     def _list_user_projects(self, user_name):
         # http://copr.fedorainfracloud.org/api/coprs/ignatenkobrain/
         api_path = "/api/coprs/{}/".format(user_name)
```

Reply:
Filenames should play no role in the logic. Just repo IDs as the single source of thruth. So I need to be -1 here.